File: src/com/dafttech/terra/game/TimeKeeping.java
package com.dafttech.terra.game;

public class TimeKeeping {

    
    static long lastTime, currentTime;
    static String lastMsg;
    
    public static void timeKeeping(String msg) {
        long temp = currentTime;
        currentTime = System.currentTimeMillis();
        if(currentTime - lastTime > 25) {
            System.out.println("Time from " + lastMsg + " to " + msg + ": " + (currentTime - lastTime));
        }
        lastMsg = msg;
        lastTime = temp;
    }
}

File: terra/src/main/scala/com/dafttech/terra/game/TimeKeeping.scala
package com.dafttech.terra.game

object TimeKeeping {
  private[game] var lastTime: Long = 0L
  private[game] var currentTime: Long = 0L
  private[game] var lastMsg: String = null

  def timeKeeping(msg: String) {
    val temp: Long = currentTime
    currentTime = System.currentTimeMillis
    if (currentTime - lastTime > 25) {
      System.out.println("Time from " + lastMsg + " to " + msg + ": " + (currentTime - lastTime))
    }
    lastMsg = msg
    lastTime = temp
  }
}